conan/cli/commands/compare.py

```diff
@@ -0,0 +1,233 @@
+        full_ref, matching_remote = None, None
+        for remote in enabled_remotes:
+            if ref.revision:
+                no_rrev_ref = RecipeReference().loads(reference)
+                no_rrev_ref.revision = None
+                try:
+                    remote_revisions = conan_api.list.recipe_revisions(no_rrev_ref, remote)
+                    if ref in remote_revisions:
+                        full_ref = ref
+                        matching_remote = remote
+                        break
+                except:
+                    continue
+            else:
+                try:
+                    latest_recipe_revision = conan_api.list.latest_recipe_revision(ref, remote)
+                except:
+                    continue
+                if full_ref is None or (latest_recipe_revision.timestamp > full_ref.timestamp):
+                    full_ref = latest_recipe_revision
+                    matching_remote = remote
+        if full_ref is None or matching_remote is None:
+            raise ConanException(f"No matching reference for {reference} in remotes")
+
+        conan_api.download.recipe(full_ref, matching_remote)
+        cache_path = conan_api.cache.export_path(ref)
+        return ref, cache_path
+
+    def _export_recipe_from_path(path_to_conanfile, reference):
+        path = conan_api.local.get_conanfile_path(path_to_conanfile, cwd, py=True)
+        ref = RecipeReference().loads(reference)
+        export_ref, conanfile = conan_api.export.export(path=path,
+                                                        name=ref.name, version=ref.version,
+                                                        user=ref.user, channel=ref.channel,
+                                                        lockfile=None,
+                                                        remotes=enabled_remotes)
+        cache_path = conan_api.cache.export_path(export_ref)
+        return export_ref, cache_path
+
+    def _source(path_to_conanfile, reference, required_ref):
+        if required_ref is not None:
+            export_ref, cache_path = _download_ref_from_remote(required_ref)
+        else:
+            export_ref, cache_path = _export_recipe_from_path(path_to_conanfile, reference)
+        exported_path = conan_api.local.get_conanfile_path(cache_path, cwd, py=True)
+        conan_api.local.source(exported_path, name=export_ref.name, version=str(export_ref.version), user=export_ref.user,
+                               channel=export_ref.channel, remotes=enabled_remotes)
+        return export_ref, cache_path
+
+    old_export_ref, old_cache_path = _source(args.old_path, args.old_reference, args.old_require)
+    new_export_ref, new_cache_path = _source(args.new_path, args.new_reference, args.new_require)
+
+
+    with open(f"diff{old_export_ref.version}-{new_export_ref.version}.patch", "w") as f:
+        subprocess.run(["git", "diff", old_cache_path, new_cache_path], stdout=f, check=False)
```

Comment:
Isn't it possible to collect the diffs in memory, and not opening a file? Like ``.run(capture_output=True)``?

Reply:
The current approach (under a TODO to find a better way) is using `check_output_runner` but that's in the `internal` module, and its functionality is not exposed anywhere public. Any suggestions?